### PR TITLE
[Core] Extending fast_transfer_between_model_parts_process

### PR DESCRIPTION
**Pull request by loumalouomega**
Adding possibility to clone entities

### CODE REVIEW

File: kratos/processes/fast_transfer_between_model_parts_process.cpp [hunk 1]
```diff
@@ -53,91 +61,282 @@ void FastTransferBetweenModelPartsProcess::Execute()
...
+            #pragma omp for
+            for(int i = 0; i < num_conditions; ++i) {
+                auto it_cond = mrOriginModelPart.ConditionsBegin() + i;
+                Condition::Pointer p_new_cond = it_cond->Clone(total_num_conditions + i + 1, it_cond->GetGeometry());
+                (conditions_buffer[id]).insert((conditions_buffer[id]).begin(), p_new_cond);
             }
         }
+
+        // We add to the model part
+        #pragma omp single
```
Comment (roigcarlo) [inline kratos/processes/fast_transfer_between_model_parts_process.cpp:252]: You may want to declare 3 different single regions so they can be executed in parallel if there are enough cores

File: kratos/processes/fast_transfer_between_model_parts_process.cpp [hunk 2]
```diff
@@ -53,91 +61,282 @@ void FastTransferBetweenModelPartsProcess::Execute()
...
+                auto it_cond = mrOriginModelPart.ConditionsBegin() + i;
+                if (it_cond->Is(mFlag)) {
+                    Condition::Pointer p_new_cond = it_cond->Clone(total_num_conditions + i + 1, it_cond->GetGeometry());
+                    (conditions_buffer[id]).insert((conditions_buffer[id]).begin(), p_new_cond);
+                }
+            }
+        }
+
+        // We add to the model part
+        #pragma omp single
```
Comment (roigcarlo) [inline kratos/processes/fast_transfer_between_model_parts_process.cpp:337]: Same

File: kratos/processes/fast_transfer_between_model_parts_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/processes/fast_transfer_between_model_parts_process.cpp:252]: Ok, wi will solve it

**Review by roigcarlo — APPROVED** [review]
Good to me

**Pull request merged**